exporters/cmake/OpenSSLConfig.cmake.in

```diff
@@ -91,11 +91,13 @@ get_filename_component(_ossl_prefix "${CMAKE_CURRENT_LIST_FILE}" PATH)
 {-
   # For each component in $OpenSSL::safe::installdata::CMAKECONFIGDIR relative to
   # $OpenSSL::safe::installdata::PREFIX, have CMake figure out the parent directory.
-  my $d = join('/', unixify($OpenSSL::safe::installdata::LIBDIR_REL_PREFIX),
-                    unixify($OpenSSL::safe::installdata::CMAKECONFIGDIR_REL_LIBDIR));
+  my $d = join('/', unixify(catdir($OpenSSL::safe::installdata::LIBDIR_REL_PREFIX,
+                                   $OpenSSL::safe::installdata::CMAKECONFIGDIR_REL_LIBDIR), 1));
   $OUT = '';
-  $OUT .= 'get_filename_component(_ossl_prefix "${_ossl_prefix}" PATH)' . "\n"
-      foreach (split '/', $d);
+  if ($d ne '.') {
+      $OUT .= 'get_filename_component(_ossl_prefix "${_ossl_prefix}" PATH)' . "\n"
+          foreach (split '/', $d);
```

Comment:
But I'm curious about the `for (...)` here, which uses `$_` presumably as the iteration variable, which does not appear to be used (both originally, and in the improved code). How does this work?

Reply:
It is a statement modifier, which is also why it's indented like that compared to the line before (the statement it modifies).

https://perldoc.perl.org/perlsyn#Statement-Modifiers

Reply:
Ah, never mind, that's not what you asked.  Yeah, `$_` isn't used, the `foreach` loop is simple there to count the number of times that same CMake statement should be added to `$OUT`, based on the number of directories in `$d`.

Reply:
I could actually have this instead:

``` perl
      $OUT .= 'get_filename_component(_ossl_prefix "${_ossl_prefix}" PATH)' . "\n"
          x scalar (split '/', $d);
```

I hope that every reader knows the `x` operator 😉